sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaIO.java

```diff
@@ -0,0 +1,307 @@
+          .setQueuePollTime(Read.DEFAULT_QUEUE_POLL_TIME)
+          .setQueueMaxPollTime(Read.DEFAULT_QUEUE_MAX_POLL_TIME)
+          .build();
+    }
+
+    /**
+     * Returns a new transform which will use the custom TikaConfig.
+     */
+    public Read withTikaConfigPath(String tikaConfigPath) {
+      checkNotNull(tikaConfigPath, "TikaConfigPath cannot be empty.");
+      return withTikaConfigPath(StaticValueProvider.of(tikaConfigPath));
+    }
+
+    /** Same as {@code with(tikaConfigPath)}, but accepting a {@link ValueProvider}. */
+    public Read withTikaConfigPath(ValueProvider<String> tikaConfigPath) {
+      checkNotNull(tikaConfigPath, "TikaConfigPath cannot be empty.");
+      return toBuilder()
+          .setTikaConfigPath(tikaConfigPath)
+          .build();
+    }
+
+    /**
+     * Returns a new transform which will use the provided content type hint
+     * to make the file parser detection more efficient.
+     */
+    public Read withContentTypeHint(String contentType) {
+      checkNotNull(contentType, "ContentType cannot be empty.");
+      Metadata metadata = new Metadata();
+      metadata.add(Metadata.CONTENT_TYPE, contentType);
+      return withInputMetadata(metadata);
+    }
+
+    /**
+     * Returns a new transform which will use the provided input metadata
+     * for parsing the files.
+     */
+    public Read withInputMetadata(Metadata metadata) {
+      Metadata inputMetadata = this.getInputMetadata();
+      if (inputMetadata != null) {
+        for (String name : metadata.names()) {
+            inputMetadata.set(name, metadata.get(name));
+        }
+      } else {
+        inputMetadata = metadata;
+      }
+      return toBuilder().setInputMetadata(inputMetadata).build();
+    }
+
+    /**
+     * Returns a new transform which will report the metadata.
+     */
+    public Read withReadOutputMetadata(Boolean value) {
+      return toBuilder().setReadOutputMetadata(value).build();
+    }
+
+    /**
+     * Returns a new transform which will use the specified queue poll time.
+     */
+    public Read withQueuePollTime(Long value) {
+      return toBuilder().setQueuePollTime(value).build();
+    }
+
+    /**
+     * Returns a new transform which will use the specified queue max poll time.
+     */
+    public Read withQueueMaxPollTime(Long value) {
+      return toBuilder().setQueueMaxPollTime(value).build();
+    }
+
+    /**
+     * Returns a new transform which will operate on the text blocks with the
+     * given minimum text length.
+     */
+    public Read withMinimumTextlength(Integer value) {
+      return toBuilder().setMinimumTextLength(value).build();
+    }
+
+    /**
+     * Returns a new transform which will use the synchronous reader.
+     */
+    public Read withParseSynchronously(Boolean value) {
+      return toBuilder().setParseSynchronously(value).build();
+    }
+
+    /**
+     * Path to Tika configuration resource.
+     */
+    public Read withOptions(TikaOptions options) {
```

Comment:
use of PipelineOptions here is not a good idea. PipelineOptions are a global object for the entire pipeline - with this method, you can't have several TikaIO's in a pipeline configured differently.

Reply:
Ok...